src/app/[locale]/_utils/api-utils.ts

```diff
@@ -1,19 +1,36 @@
 /* eslint-disable no-console */
+import { URL } from 'url';
 import { agent } from '@pagopa/ts-commons';
-import { AbortableFetch, setFetchTimeout, toFetch } from '@pagopa/ts-commons/lib/fetch';
 import { ApiRequestType, IResponseType, TypeofApiResponse } from '@pagopa/ts-commons/lib/requests';
 import { Millisecond } from '@pagopa/ts-commons/lib/units';
 import { isRight, toError } from 'fp-ts/lib/Either';
 import * as t from 'io-ts';
+import * as E from 'fp-ts/Either';
+import * as TE from 'fp-ts/TaskEither';
+import { pipe } from 'fp-ts/function';
+import { calculateExponentialBackoffInterval } from '@pagopa/ts-commons/lib/backoff';
+import {
+  AbortableFetch,
+  retriableFetch,
+  setFetchTimeout,
+  toFetch,
+} from '@pagopa/ts-commons/lib/fetch';
+import { RetriableTask, TransientError, withRetries } from '@pagopa/ts-commons/lib/tasks';
 import { ROUTES } from './routes';
+//
+// Returns a fetch wrapped with timeout and retry logic
+//
+const API_TIMEOUT = Number(`${process.env.NEXT_PUBLIC_API_FETCH_TIMEOUT}`) as Millisecond;
+const API_MAX_RETRY = Number(`${process.env.NEXT_PUBLIC_API_FETCH_MAX_RETRY}`);
 
 /** Return the implementation of fetch configured with a timeout */
-export const buildFetchApi = (
-  timeoutMs: number = 300000
```

Comment:
Why not leaving timeoutMs as parameter defaulting to `API_TIMEOUT`?

Reply:
Leaving now **timeoutMs** as parameter default